core/store.go

```diff
@@ -290,6 +291,111 @@ func Get(k string) *Obj {
 	return getHelper(k, true)
 }
 
+var (
+	keyList []string
+)
+
+// initKeyList initializes keyList with keys from keypool if it is empty or outdated.
+// This function is used to populate keyList with all keys from keypool,
+// ensuring that it matches the current state of the keypool.
+func initKeyList() {
+	// Check if keyList is empty or needs to be updated
+	if len(keyList) == 0 || len(keyList) != len(keypool) {
```

Comment:
len(keyList) != len(keypool)
Will this work in all cases? What if the number of keys that are added/deleted are same?
What happens if a number of keys expire at the same time?

Reply:
No, it won't work I guess because it, but will need to check after the keylist scope changed to client specific, and also need to rethink about how to make keylist computation efficient as currently I am calling init on each scan.